connectors/sources/google_cloud_storage.py

```diff
@@ -199,10 +201,40 @@ async def _api_call(
                 )
                 await asyncio.sleep(DEFAULT_WAIT_MULTIPLIER**retry_counter)
 
+    def get_service_account_credentials(self):
+        """Initialize and return the ServiceAccountCreds
+
+        Returns:
+            ServiceAccountCreds: An instance of the ServiceAccountCreds.
+        """
+        if self.service_account_credentials is not None:
+            return
+
+        json_credentials = json.loads(self.configuration["service_account_credentials"])
+
+        if (
+            json_credentials.get("private_key")
+            and "\n" not in json_credentials["private_key"]
+        ):
+            json_credentials["private_key"] = get_pem_format(
+                key=json_credentials["private_key"].strip(),
+                max_split=2,
+            )
+
+        self.service_account_credentials = ServiceAccountCreds(
+            scopes=[CLOUD_STORAGE_READ_ONLY_SCOPE],
+            **json_credentials,
+        )
+        self.user_project_id = self.service_account_credentials.project_id
+        return self.service_account_credentials
+
     async def ping(self):
         """Verify the connection with Google Cloud Storage"""
+
         if RUNNING_FTEST:
             return
+
+        self.get_service_account_credentials()
```

Comment:
I had a chat with @praveen-elastic about that. I think we should encapsulate instantiating a session/client in a separate class, which provides methods for talking with the corresponding API. Because here we call a getter and ignore the result and rely on the internal side effect, that `self.service_account_credentials` is now instantiated, which is not obvious.

My suggestion would be something like the following:
(This is only a suggestion, you can choose another name, if you think it's more appropriate)
```
class GoogleCloudStorageClient:
    def __init__()
        self.service_account_credentials = ...
        # you can also store a client instance here        

    def api_call(self, resource):
        # execute api call

class GoogleCloudStorageDataSource(BaseDataSource):
     def get_client(self):
         if self.client is not None:
             return self.client
         self.client = GoogleCloudStorageClient(...)
         
         return self.client 
```